testing/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/NebulaPluginsSmokeTest.groovy

```diff
@@ -246,3 +257,30 @@ testImplementation('junit:junit:4.7')""")
         ]
     }
 }
+
+class NebulaPluginDeprecations extends BaseDeprecations {
```

Comment:
💅 Move it to its own file

Reply:
Moved as a static inner class, as seem to be the common approach for other `BaseDeprecation` subtypes (except for a couple of unused ones)